test/test_cuda.py

```diff
@@ -5770,6 +5770,189 @@ def test_no_triton_on_import(self):
         self.assertEqual(rc, "False", "Triton was imported when importing torch!")
 
 
+@unittest.skipIf(IS_WINDOWS or TEST_WITH_ROCM, "Not compatible with windows or ROCm")
+class TestCUDAPluggableAllocator(TestCase):
+    def setUp(self):
+        super().setUp()
+        self._template = """\
+import torch
+from setuptools import setup
+from torch.utils import cpp_extension
+
+custom_allocator_code = '''
+#include <sys/types.h>
+#include <cuda_runtime_api.h>
+#include <torch/extension.h>
+
+extern "C" {{
+void* my_malloc(ssize_t size, int device, cudaStream_t stream) {{
+   {malloc}
+}}
+
+void my_free(void* ptr, ssize_t size, int device, cudaStream_t stream) {{
+   {free}
+}}
+}}
+
+// Needed for torch extension to work
+PYBIND11_MODULE(TORCH_EXTENSION_NAME, m) {{}}
+'''
+module = cpp_extension.load_inline(
+    name='{name}',
+    cpp_sources=custom_allocator_code,
+    build_directory='.',
+    with_cuda=True,
+)
```

Comment:
I got the same error locally as in CI: no .so is generated because you don't ask for any functions.

Reply:
This is weird, I see all the .so functions generated in my local env ...
Could this be compiler versions? or the path where the compiled module is being stored?

Reply:
after building them in a tempdir this was solved :)